onnx/defs/tensor/utils.cc

```diff
@@ -54,7 +91,36 @@ void resizeShapeInference(InferenceContext& ctx) {
   }
   const auto& input_shape = getInputShape(ctx, 0);
   auto* output_shape = getOutputShape(ctx, 0);
+
+  bool hasScalesInput = (2 < ctx.getNumInputs()) && (ctx.getInputType(2) != nullptr);
+  bool hasSizesInput = (3 < ctx.getNumInputs()) && (ctx.getInputType(3) != nullptr);
+
+  if (hasScalesInput + hasSizesInput != 1) {
+    fail_shape_inference("Either `sizes` or `scales` must be provided, but not both of them");
```

Comment:
FYI this line causes shape inference failures with some of ONNX Model Zoo models. Please check this issue: https://github.com/onnx/onnx/issues/4380. Thanks!